airgun/views/redhat_repository.py

```diff
@@ -0,0 +1,271 @@
+    LABEL = ".//div[contains(@class, 'item-text')]"
+    ITEMS = ".//div[contains(@class, 'list-item-with-divider')]"
+
+    @property
+    def expand_button(self):
+        """Return the expand button element."""
+        return self.browser.element(self.EXPAND_BUTTON, parent=self)
+
+    @property
+    def expanded(self):
+        """Check whether this repository set is expanded or not."""
+        return 'active' in self.browser.get_attribute('class', self.expand_button)
+
+    def expand(self):
+        """Expand the repository set item section."""
+        if not self.expanded:
+            self.browser.click(self.EXPAND_BUTTON, parent=self)
+            _wait_for_spinner(self.parent)
+
+    @property
+    def name(self):
+        """Return the name displayed for this repository item"""
+        return self.browser.text(self.NAME, parent=self)
+
+    @property
+    def label(self):
+        """Return the label displayed for this repository item"""
+        return self.browser.text(self.LABEL, parent=self)
+
+    @property
+    def items(self):
+        """Return all the items (available repositories) of this repository set."""
+        if self.ITEM:
+            self.expand()
+            return [
+                AvailableRepositoryItem(self, '{0}[{1}]'.format(self.ITEMS, index + 1))
+                for index, _ in enumerate(self.browser.elements(self.ITEMS, parent=self))
+            ]
+        return []
+
+    def read_items(self):
+        """Read all the items (repositories) of this repository set"""
+        return [item.text for item in self.items]
+
+    def read(self):
+        """Return the name and label of this repository."""
+        return dict(name=self.name, label=self.label)
+
+    def enable(self, item):
+        """Enable a repository of this repository set.
+
+        :param str item: The arch and version (if available) of the repository.
+        """
+        self.expand()
+        for arch_version_item in self.items:
+            if item == arch_version_item.text:
+                arch_version_item.enable()
+                break
+        else:
+            raise ValueError(
+                'Repository "{0}" was not found in repository set "{1}"'.format(item, self.name))
+
+    def fill(self, value):
+        """Fill this widget with value, this will enable the repository name in this repository set
+            if value is True."""
+        if value is True:
+            self.enable(value)
+
+
+class EnabledRepositoryWidget(AvailableRepositorySetWidget):
+    """The widget representation of Enabled repository item."""
+    ITEM = None
+    DISABLE_BUTTON = ".//button"
+
+    def disable(self):
+        """Disable this repository."""
+        self.browser.click(self.DISABLE_BUTTON, parent=self)
+        _wait_for_spinner(self.parent)
+
+    def fill(self, value):
+        """Fill this widget with value, this will disable this repository if value is False."""
+        if value is False:
```

Comment:
>maybe we should rise some exception if value is smth else?

Reply:
I think we should not be explicit here. if False disable , if not ignore, that's what I think, this does not worth additional work from the user.

Reply:
not sure that I get the logic, it is still like:
```
def fill(self, value):
    if value is True:
        self.disable()
```

Reply:
@oshtaier Sorry I did not get your point here.